File: 2015/22/22.js
var word = "evdhtiqgfyvcytohqppcmdbultbnzevdbakvkcdpbatbtjlmzaolfqfqjifkoanqcznmbqbeswglgrzfroswgxoritbw";

var operations = 0;
for (var i = 0; i < word.length/2; i++) {
	operations += Math.abs(word.charCodeAt(i) - word.charCodeAt(word.length-i-1));
}

console.log(operations);

File: 2015/22/22.hs
import Data.Char (ord)

reduceToPalindrome :: String -> Int
reduceToPalindrome str = sum $ zipWith (\c1 c2 -> abs((ord c1) - (ord c2))) s r
    where halfLength = (length str) `div` 2
          s = take halfLength str
          r = take halfLength (reverse str)

main = putStrLn $ show $ reduceToPalindrome "evdhtiqgfyvcytohqppcmdbultbnzevdbakvkcdpbatbtjlmzaolfqfqjifkoanqcznmbqbeswglgrzfroswgxoritbw"
